core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -249,6 +249,9 @@ private[spark] class EventLoggingListener(
   }
 
   override def onExecutorMetricsUpdate(event: SparkListenerExecutorMetricsUpdate): Unit = {
+    if (event.execId == SparkContext.DRIVER_IDENTIFIER) {
+      logEvent(event)
+    }
```

Comment:
Do this only when `shouldLogStageExecutorMetrics` is enabled ?

Reply:
> Do this only when `shouldLogStageExecutorMetrics` is enabled ?

I don't thinks it should be controlled by `shouldLogStageExecutorMetrics `. since driver's metrics is not related to stage  executor.

Reply:
Currently, we have a single event for both driver and executor metrics update - differentiated by exec id.
I dont have strong opinions on this, but if we have a flag (`shouldLogStageExecutorMetrics`) controlling whether metrics are to be updated, we should consistently apply it IMO.

Reply:
> Currently, we have a single event for both driver and executor metrics update - differentiated by exec id.
> I dont have strong opinions on this, but if we have a flag (`shouldLogStageExecutorMetrics`) controlling whether metrics are to be updated, we should consistently apply it IMO.

@mridulm Follow this comment, how about current.